src/Build/Evaluation/ProjectStringCache.cs

```diff
@@ -84,6 +85,13 @@ internal int DocumentCount
         /// </summary>
         public string Add(string key, XmlDocument document)
         {
+            // Remove string interning in ChangeWave 17.4
+            // Note: When ready to remove the ChangeWaves under 17.4, please remove all related and no more used code (see the PR https://github.com/dotnet/msbuild/pull/7952).
```

Comment:
```suggestion
            // Note: When ready to remove the ChangeWaves under 17.4, please delete this entire class and all references to it. (See the PR https://github.com/dotnet/msbuild/pull/7952).
```

We can also delete unused methods here now. Looks like there are no references to Count or DocumentCount, for instance.

Reply:
I thought about not having references in some places of the code, I also noticed that. No, that's for tests and it is used via Reflections, so it is not yet dead code.